reconcile/utils/rest_api_base.py

```diff
@@ -0,0 +1,92 @@
+import threading
+from typing import Any
+from urllib.parse import urljoin
+
+import requests
+
+
+def get_next_url(links: dict[str, dict[str, str]]) -> str | None:
+    """Parse response header 'Link' attribute and return the next page url if exists.
+
+    See
+    * https://gitlab.com/glitchtip/glitchtip-backend/-/blob/master/glitchtip/pagination.py#L34
+    * https://requests.readthedocs.io/en/latest/api/?highlight=links#requests.Response.links
+    """
+    if links.get("next", {}).get("results", "false") == "true":
+        return links["next"]["url"]
+    return None
+
+
+class ApiBase:
+    """This class provides a common standard for REST API clients."""
+
+    def __init__(
+        self,
+        host: str,
+        token: str,
+        max_retries: int | None = None,
+        read_timeout: float | None = None,
+    ) -> None:
+        self.host = host
+        self.token = token
+        self.max_retries = max_retries if max_retries is not None else 3
+        self.read_timeout = read_timeout if read_timeout is not None else 30
+        self._thread_local = threading.local()
```

Comment:
now we are making this as a base, let's fix the session issue to make it use the same session object and do auto close with context manager.

Reply:
Yes, you're right. Changed that in the last commit.